x-pack/plugin/security/src/test/java/org/elasticsearch/integration/IndexPrivilegeTests.java

```diff
@@ -385,6 +391,18 @@ public void testUserU14() throws Exception {
                 "GET", "/" + randomIndex() + "/_mtermvectors", "{ \"docs\" : [ { \"_id\": \"1\" }, { \"_id\": \"2\" } ] }");
     }
 
+    public void testUser15() throws Exception {
+        assertUserIsAllowed("u15", "maintenance", "a");
+
+        assertAccessIsAllowed("u15",
+            "GET", "/" + randomIndex() + "/_msearch", "{}\n{ \"query\" : { \"match_all\" : {} } }\n");
+        assertAccessIsAllowed("u15", "POST", "/" + randomIndex() + "/_mget", "{ \"ids\" : [ \"1\", \"2\" ] } ");
+        assertAccessIsDenied("u15", "PUT",
+            "/" + randomIndex() + "/_bulk", "{ \"index\" : { \"_id\" : \"123\" } }\n{ \"foo\" : \"bar\" }\n");
+        assertAccessIsAllowed("u15",
+            "GET", "/" + randomIndex() + "/_mtermvectors", "{ \"docs\" : [ { \"_id\": \"1\" }, { \"_id\": \"2\" } ] }");
```

Comment:
I don't think we need this fluff. This is testing the authorization of composite requests but refresh and flush are not part of those composite type of requests.
In addition to `assertUserIsAllowed("u15", "maintenance", "a");` I would also check that `crud` is not allowed, and we should also test that other users in this class don't have the `maintenance` priv (right now it only checks the positive case).